packages/dds/map/package.json

```diff
@@ -12,12 +12,14 @@
   "types": "dist/index.d.ts",
   "scripts": {
     "build": "npm run build:genver && concurrently npm:build:compile npm:lint",
-    "build:compile": "concurrently npm:tsc npm:build:esnext",
+    "build:commonjs": "npm run tsc && npm run build:test",
```

Comment:
>: "npm run tsc && npm run build:test", [](start = 20, length = 38)

will this result in the src being built twice?

Reply:
should we be using build mode to build both project, rather than two tsc command: [https://www.typescriptlang.org/docs/handbook/project-references.html#build-mode-for-typescript](https://www.typescriptlang.org/docs/handbook/project-references.html#build-mode-for-typescript)

---
In reply to: [532981773](https://github.com/microsoft/FluidFramework/pull/4494#discussion_r532981773) [](ancestors = 532981773)

Reply:
When it is separate the `package.json`, the `fluid-build` tool can schedule them and prioritize the src directory because of it's detection, which unblock building it's dependencies.   There is probably a small benefit there as we add more test files and building tests takes more time, although using the `-b` options might reduce the overhead of extra process invocation.   I can do some measurement either way after I have move everything into place.

Reply:
i like the idea of our package dirs remaining self contained, so using package refs, and -b on the tsc command makes sense, rather than depending on fluid-buid to do the right things at the mono-repo level. 

At least on windows the processs overhead can be pretty high.

Reply:
The package.json `npm run build` script works to build both as well, so it is self contained.